apis/namespaced/apigee/v1beta1/zz_envreferences_types.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Code generated by upjet. DO NOT EDIT.
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	v1 "github.com/crossplane/crossplane-runtime/v2/apis/common/v1"
+	v2 "github.com/crossplane/crossplane-runtime/v2/apis/common/v2"
+)
+
+type EnvReferencesInitParameters struct {
+	Description *string `json:"description,omitempty" tf:"description,omitempty"`
+
+	Refers *string `json:"refers,omitempty" tf:"refers,omitempty"`
+
+	ResourceType *string `json:"resourceType,omitempty" tf:"resource_type,omitempty"`
+}
+
+type EnvReferencesObservation struct {
+	Description *string `json:"description,omitempty" tf:"description,omitempty"`
+
+	EnvID *string `json:"envId,omitempty" tf:"env_id,omitempty"`
+
+	ID *string `json:"id,omitempty" tf:"id,omitempty"`
+
+	Refers *string `json:"refers,omitempty" tf:"refers,omitempty"`
+
+	ResourceType *string `json:"resourceType,omitempty" tf:"resource_type,omitempty"`
+}
+
+type EnvReferencesParameters struct {
+
+	// +kubebuilder:validation:Optional
+	Description *string `json:"description,omitempty" tf:"description,omitempty"`
+
+	// +kubebuilder:validation:Required
+	EnvID *string `json:"envId" tf:"env_id,omitempty"`
```

Comment:
Not a blocker but for better XRM compliancy: Again, can we make this a reference to an Apigee environment resource? Is it also possible to specify an environment group here as the docs mention about groups but the expected target resource name syntax belongs to environments and not groups?:
```
env_id - (Required) The Apigee environment group associated with the Apigee environment, in the format organizations/{{org_name}}/environments/{{env_name}}.
```

Another complication is, if I'm not mistaken, we still do not generate CEL rules that will require one of the three ref. resolver fileds (either the target directly, or the ref or selector fields). So, all these three fields become optional. But the `spec.forProvider.envID` is a required field.